packages/spacecat-shared-rum-api-client/src/functions/form-vitals.js

```diff
@@ -143,17 +198,25 @@ function handler(bundles) {
   const formViewdataChunks = new DataChunks();
   loadBundles(bundlesWithFilteredEvents, formViewdataChunks);
   const formSourceMap = {};
+  const iframeParentMap = {};
   const globalFormSourceSet = new Set();
-  formViewdataChunks.filter = { checkpoint: ['viewblock'] };
+  formViewdataChunks.filter = { checkpoint: ['viewblock', 'viewmedia'] };
   formViewdataChunks.filtered.forEach(({ url, events }) => {
     formSourceMap[url] = formSourceMap[url] || new Set();
-    events.forEach(({ checkpoint, source }) => {
+    events.forEach(({ checkpoint, source, target }) => {
       if (checkpoint === 'viewblock' && source) {
         formSourceMap[url].add(source);
         globalFormSourceSet.add(source);
       }
+      if (checkpoint === 'viewmedia' && target) {
+        if (target.includes('iframe')) {
+          iframeParentMap[target] = iframeParentMap[target] || new Set();
+          iframeParentMap[target].add(generateKey(url, source));
```

Comment:
the source is of the iframe. we'd need source of the form that is in the iframe if we want to use this value in `formsource` attribute.